cr.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Netguru iOS code review task
+//
+
+class paymentViewController: UIViewController {
+    var delegate: PaymentViewControllerDelegate
+    let customView = PaymentView()
+    let payment: Payment?
+
+    func callbacks() {
+        view.backgroundColor = UIColor(displayP3Red: 1, green: 1, blue: 1, alpha: 1)
+        customView.didTapButton = {
+            if let payment = self.payment {
+                self.delegate.didFinishFlow(amount: payment.amount
+                                            currency: payment.currency)
+            }
+        }
```

Comment:
We can make better job of this dependency by making it more explicit and requiring in view init like in example below:

``` swift
final class PaymentView: UIView {
    
    private let didTapButton: () -> Void
    
    init(didTapButton: @escaping () -> Void) {
        self.didTapButton = didTapButton
        super.init(frame: .zero)
    }
    
    // ...
    
}
```

now we make it clear that this view requires that action to be provided and we will take care of it in the `loadView`

Reply:
It's also worth to mention in here that, we should capture `self` weakly in this callback, as this will also create a strong reference cycle.